javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,131 @@
+let info = [{       // User name array
+    name: ""
+}]
+
+let todolist = []       // Todo list array
+
+function getReply(command) {
+    let arr = []        // Users inputs arrays
+    command1 = command.split(' ')
+    arr.push(command1)
+
+    switch (arr[0][0] + arr[0][1] + arr[0][2] + arr[0][3]) {
+
+        // Introduce inputs
+
+        case 'Hellomynameis':
+            if (info[0].name == arr[0][4]) {
+                console.log(`Okey okey i got it. Your name is ${info[0].name}. You dont need to repeat that.`)
+            }
+            else {
+                info[0].name = arr[0][4]
+                console.log(`Nice to meet you ${info[0].name}`)
+            }
+            break
+
+        // User name
+
+        case 'Whatismyname':
+            if (info[0].name == "") {
+                console.log('How can i know that. You dont introduce yourself!!!')
+            }
+            else {
+                console.log(`Your name is ${info[0].name}`)
+            }
+            break
+
+        // Set Timer
+
+        case 'Setatimerfor':
+            if (arr[0][5] == 'minutes') {
+                time = arr[0][4] * 60000
+            }
+            else if (arr[0][5] == 'seconds') {
+                time = arr[0][4] * 1000
+            }
+            else if (arr[0][5] == 'hours') {
+                time = arr[0][4] * 3600000
+            }
+            else { console.log('You write to wrong timer!!!') }
+             console.log(`Timer set for ${arr[0][4]} ${arr[0][5]}`)
+            setTimeout(function () {
+                console.log('Timer Done!!!');
+            }, time);
+            break
+
+        // Random number
+
+        case 'Sayanumberbetween' :
+            let min = Number(arr[0][4])
+            let max = Number(arr[0][6])
+            let random = min + Math.random() * (max-min)
+            console.log(Math.floor(random))
+    }
+
+    // Another switch function
+
+    lngth = arr[0].length
+
+    switch (arr[0][0] + arr[0][lngth - 3] + arr[0][lngth - 2] + arr[0][lngth - 1]) {
+
+        // Add to todo list
+
+        case 'Addtomytodo':
+            todolist.push(arr[0].slice(1, lngth - 3))
```

Comment:
`.slice` returns an array, which means `todolist` ends up being an array with an array, which is unnecessarily complicated.
I'd assign these variables so it becomes easier to read:
```js
// lets say the command was "Add walking the dog to my todo"
const newTodoItem = commandWords.slice(1, length - 3).join(" "); // this will set newTodoItem to "walking the dog"
todolist.push(newTodoItem);

console.log(`${newTodoItem} added to your todo`);
```